libr/anal/p/anal_pyc.c

```diff
@@ -0,0 +1,166 @@
+/* radare - LGPL3 - Copyright 2016-2020 - FXTi */
+
+#include <r_types.h>
+#include <r_lib.h>
+#include <r_util.h>
+#include <r_asm.h>
+
+#include "../../asm/arch/pyc/opcode.h"
+#include "../../asm/arch/pyc/pyc_dis.h"
+
+static pyc_opcodes *ops = NULL;
+
+static int archinfo(RAnal *anal, int query) {
+	if (!strcmp (anal->cpu, "x86"))
+		return -1;
+
+	switch (query) {
+	case R_ANAL_ARCHINFO_MIN_OP_SIZE:
+		return (anal->bits == 16) ? 1 : 2;
+	case R_ANAL_ARCHINFO_MAX_OP_SIZE:
+		return (anal->bits == 16) ? 3 : 2;
+	default:
+		return -1;
+	}
+}
+
+static char *get_reg_profile(RAnal *anal) {
+	return strdup (
+		"=PC    pc\n"
+		"=BP    bp\n"
+		"=SP    sp\n"
+		"gpr    sp  .32 0   0\n" // stack pointer
+		"gpr    pc  .32 4   0\n" // program counter
+		"gpr    bp  .32 8   0\n" // base pointer // unused
+	);
+}
+
+static RList *get_pyc_code_obj(RAnal *anal) {
+	RBin *b = anal->binb.bin;
+	RBinPlugin *plugin = b->cur && b->cur->o ? b->cur->o->plugin : NULL;
+	ut8 is_pyc = (plugin && strcmp (plugin->name, "pyc") == 0) ? 1 : 0;
+	return is_pyc ? b->cur->o->bin_obj : NULL;
+}
+
+static int pyc_op(RAnal *a, RAnalOp *op, ut64 addr, const ut8 *data, int len, RAnalOpMask mask) {
+	RList *cobjs = r_list_get_n (get_pyc_code_obj (a), 0);
+	RListIter *iter = NULL;
+	pyc_code_object *func = NULL, *t = NULL;
+	r_list_foreach (cobjs, iter, t) {
+		if (t->start_offset <= addr && addr < t->end_offset) { // addr in [start_offset, end_offset)
```

Comment:
Imho its more readable to use addr > x && addr < y (or use the R_BETWEEN macro)

Reply:
i agree.